spec/models/project_group_spec.rb

```diff
@@ -41,18 +41,63 @@
     end
   end
 
-  describe "descendant_leaf_count" do
+  context "with descendants" do
     let(:root) { create(:root_project_group, :with_descendants) }
 
-    it "should be correct for root" do
-      expect(root.descendant_leaf_count).to eq(
-        # There is one childless group that should be counted in addition to all the steps.
-        root.project.timeline_entries.where(type: "ProjectStep").count + 1)
+    describe "descendant_leaf_count" do
+      it "should be correct for root" do
+        expect(root.descendant_leaf_count).to eq(
+          # There is one childless group that should be counted in addition to all the steps.
+          root.project.timeline_entries.where(type: "ProjectStep").count + 1)
+      end
+
+      it "should be correct for interior node" do
+        group = root.children[0]
+        expect(group.descendant_leaf_count).to eq group.children.size
+      end
     end
 
-    it "should be correct for interior node" do
-      group = root.children[0]
-      expect(group.descendant_leaf_count).to eq group.children.size
+    describe "max_descendant_group_depth" do
```

Comment:
This section I just moved from `timeline_entry_spec` (because these only apply to project groups).